File: code/Project_4/BT5/c++/problems/problem_1_3.cpp
#include <iostream>
#include <cmath>
using namespace std;

long long count_spanning_trees_complete(int n) {
    if (n <= 1) return 1;
    long long result = 1;
    for (int i = 0; i < n - 2; i++) {
        result *= n;
    }
    return result;
}

void manual_spanning_trees_example() {
    cout << "Graph: Triangle (3 vertices, 3 edges)" << endl;
    cout << "Spanning trees:" << endl;
    cout << "1. Edges: (0,1) and (0,2)" << endl;
    cout << "2. Edges: (0,1) and (1,2)" << endl;
    cout << "3. Edges: (0,2) and (1,2)" << endl;
    cout << "Total: 3 spanning trees" << endl;
}

int main() {
    manual_spanning_trees_example();
    
    cout << "\nComplete Graphs - Number of Spanning Trees:" << endl;
    for (int n = 1; n <= 5; n++) {
        long long count = count_spanning_trees_complete(n);
        cout << "K" << n << ": " << count << " spanning trees" << endl;
    }
    
    return 0;
}

File: code/Project_4/BT5/python/prolems/problem_1_3.py
def count_spanning_trees_complete(n):
    """Số cây khung của Kn sử dụng công thức Cayley: n^(n-2)"""
    if n <= 1:
        return 1
    result = 1
    for i in range(n - 2):
        result *= n
    return result

def manual_spanning_trees_example():
    """Ví dụ thủ công cho đồ thị tam giác"""
    print("Graph: Triangle (3 vertices, 3 edges)")
    print("Spanning trees:")
    print("1. Edges: (0,1) and (0,2)")
    print("2. Edges: (0,1) and (1,2)")
    print("3. Edges: (0,2) and (1,2)")
    print("Total: 3 spanning trees")

def main():
    manual_spanning_trees_example()
    
    print("\nComplete Graphs - Number of Spanning Trees:")
    for n in range(1, 6):
        count = count_spanning_trees_complete(n)
        print(f"K{n}: {count} spanning trees")

if __name__ == "__main__":
    main()